### PR TITLE
Added list tables feature

### PR DESCRIPTION
**Pull request by josemazo**
Story [ch70779](https://app.clubhouse.io/cartoteam/story/70779/list-carto-tables)

### CODE REVIEW

**Comment by josemazo** [comment]
Ready for review again!

File: tests/unit/utils/managers/test_context_manager.py [hunk 1]
```diff
@@ -208,3 +211,37 @@ def has_table(table_name):
         # Then
         mock.assert_called_once_with('table_name', 'new_table_name')
         assert result == 'new_table_name'
+
+    def test_list_tables(self, mocker):
+        # Given
+        Dataset = namedtuple('Dataset', ['name', 'updated_at'])
+
+        mocker.patch('cartoframes.io.managers.context_manager._create_auth_client')
+        mocker.patch.object(ContextManager, 'execute_query', return_value={
```
Comment (Jesus89) [inline tests/unit/utils/managers/test_context_manager.py:220]: We could remove `execute_query` mocks right?

File: tests/unit/utils/managers/test_context_manager.py [hunk 2]
```diff
@@ -208,3 +211,37 @@ def has_table(table_name):
...
+        cm = ContextManager(self.credentials)
+        tables = cm.list_tables()
+
+        # Then
+        assert DataFrame(['table_zero', 'table_one'], columns=['tables']).equals(tables)
+
+    def test_list_tables_empty(self, mocker):
+        # Given
+        mocker.patch('cartoframes.io.managers.context_manager._create_auth_client')
+        mocker.patch.object(ContextManager, 'execute_query', return_value={
```
Comment (Jesus89) [inline tests/unit/utils/managers/test_context_manager.py:237]: Same here

File: tests/unit/utils/managers/test_context_manager.py (hunk 1 above)
Comment (josemazo) [reply tests/unit/utils/managers/test_context_manager.py]: Ups!

File: tests/unit/utils/managers/test_context_manager.py (hunk 2 above)
Comment (josemazo) [reply tests/unit/utils/managers/test_context_manager.py]: Ups (bis)!

**Pull request merged**